clang/lib/Basic/Targets/PPC.h

```diff
@@ -80,6 +80,7 @@ class LLVM_LIBRARY_VISIBILITY PPCTargetInfo : public TargetInfo {
   bool IsISA3_0 = false;
   bool IsISA3_1 = false;
   bool HasQuadwordAtomics = false;
+  bool FullRegisterNames = false;
```

Comment:
Using a target feature bit for assembly printing seems not match other bits. IIUC, all the bits here should control the available instructions on a subtarget.

Could we use an option in `TargetMachine::Options::MCOptions`? this looks like the way where clang can accept an option and control the code generation in the backend. The current option `Options.MCOptions.PreserveAsmComments` seems a little similar with this functionality.

Reply:
I did take a look at trying to do it this way at first but I was concerned that the option I wanted to add was target specific and I didn't want to add an option to `MCAsmInfo.h` that was only for PPC. However, I've taken a look at the existing options in there and it seems that there are already a set of options that are specific to one target or another so I'll try this approach.